msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+                        new UserAccountData(foreignCloud, randomUuidString(), randomUuidString())
+                    ))
+                )}
+        );
+    }
+
+    private static String randomUuidString() {
+        return UUID.randomUUID().toString();
+    }
+
+    @Before
+    public void setup() {
+        super.setup();
+        mMultipleAccountPCA = (IMultipleAccountPublicClientApplication) mApplication;
+        performInteractiveAcquireTokenWithMockResponsesToSetupCache();
+    }
+
+    @Test
+    public void testGetAccountsReturnsSingleAccountObjectForAccountRecordsFromMultipleCloudsWithSameHomeAccountId() {
+        // arrange
+        final ResultFuture<AsyncResult<List<IAccount>>> future = new ResultFuture<>();
+
+        // act
+        mMultipleAccountPCA.getAccounts(new IPublicClientApplication.LoadAccountsCallback() {
+            @Override
+            public void onTaskCompleted(List<IAccount> result) {
+                future.setResult(new AsyncResult<List<IAccount>>(result, null));
+            }
+
+            @Override
+            public void onError(MsalException exception) {
+                future.setResult(new AsyncResult<List<IAccount>>(null, exception));
+            }
+        });
+
+        RoboTestUtils.flushScheduler();
+
+        // assert
+        int expectedTenantProfilesCount = homeAccountSignedIn ?
+                testCaseData.userAccountsData.size() - 1 : testCaseData.userAccountsData.size();
+        try {
+            final AsyncResult<List<IAccount>> result = future.get();
+
+            if (result.getSuccess()) {
+                assertEquals("number of accounts", 1, result.getResult().size());
+                MultiTenantAccount account = (MultiTenantAccount) result.getResult().get(0);
+                assertEquals("number of tenant profiles",
+                        expectedTenantProfilesCount , account.getTenantProfiles().size());
+                assertTrue(testCaseData.homeAccountId.contains(account.getId()));
+                if(homeAccountSignedIn) {
+                    assertNotNull("account.getClaims()", account.getClaims());
+                    assertNotEquals("claims count",0, account.getClaims().size());
+                    verifyAccountDetails(testCaseData.userAccountsData.get(0), account);
+                }
+
+                int tenantProfileIndexStart = homeAccountSignedIn? 1: 0;
+                int tenantProfileIndexEnd = homeAccountSignedIn?
+                        expectedTenantProfilesCount: expectedTenantProfilesCount -1;
+                for(int i = tenantProfileIndexStart; i <= tenantProfileIndexEnd; i++) {
+                    verifyAccountDetails(
+                            testCaseData.userAccountsData.get(i),
+                            account.getTenantProfiles().get(testCaseData.userAccountsData.get(i).tenantId));
+                }
+            } else {
+                fail(result.getException().getMessage());
```

Comment:
Instead of doing if/else blocks for success or not, you can just assert for success: `assertTrue(result.getSuccess())`